### PR TITLE
Ignore getLegendGraphic URL if GetMap url is ignored

### PR DESCRIPTION
**Pull request by mhugent**
This PR goes has the same purpose as https://github.com/qgis/QGIS/pull/39231 , but uses a simpler approach.
In many WMS Servers, the urls advertisedin the GetCapabilities-Document are wrong or they are some internal urls not reachable from outside. In this case, it should be possible to ignore the advertised urls. Currently, the urls for GetMap and GetFeatureInfo can be ignored, but not the GetLegendGraphic url.
This PR ignores the GetLegendGraphic url, if the  GetMap url is ignored (as in PR 39231). Another possibility would be to introduce a new mIgnoreGetLegendGraphicUrl setting and add a checkbox into the WMS connection dialog.What do you think?

### CODE REVIEW

**Comment by mhugent** [comment]
Actually this more a bug fix than a new feature. I'm currently unable to get the legend  for many of my WMS layers.

**Comment by nyalldawson** [comment]
@mhugent I was tagging all potentially dangerous commits as frozen now, given that we're only a couple of days out from release. Can't this wait till 3.18.1? It's not a new regression, is it?

**Comment by mhugent** [comment]
Ok, no problem to go with 3.18.1
Actually for daily use, the 3.16 is more important to  me. Do you think it can go into 3.16.4 or 3.16.5?

**Comment by elpaso** [comment]
LGTM, I'm just wondering if the logic behind the flag is still clear for the user. Can you check if the GUI needs updates in the labels and/or tooltip to describe what exactly this flag does?

**Comment by mhugent** [comment]
Yes,true. Currently it is 'Ignore GetMap/GetTile URI reported in capabilities', it could  be 'Ignore GetMap/GetTile/GetLegendGraphic URI reported in capabilities'

**Review by elpaso — APPROVED** [review]
Let's wait for the next point release though.

**Pull request merged**
